### PR TITLE
Rest 83 - [Implementation CSV Response handle]

### PR DESCRIPTION
**Pull request by umang-clairvoyant**

### CODE REVIEW

File: restonomer-core/src/it/resources/mock_data/converter/csv/__files/sample_csv_response_body.csv [hunk 1]
```diff
@@ -0,0 +1,3 @@
+col_1,col_2,col_3
+1,2,3
+4,5,6
```
Comment (rahulbhatia023) [inline restonomer-core/src/it/resources/mock_data/converter/csv/__files/sample_csv_response_body.csv:3]: Add new line at the end of file

File: restonomer-core/src/test/scala/com/clairvoyant/restonomer/core/converter/CSVResponseToDataFrameConverterSpec.scala [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  "convertResponseToDataFrame() - with csv response body" should "return a dataframe with correct count and schema" in {
+    val responseBody =
+      """col1,col2
+        |1,2
+        |3,4""".stripMargin
+
+    val expectedSchema = "struct<col1:string,col2:string>"
+
+    val df = new CSVResponseToDataFrameConverter().convertResponseToDataFrame(responseBody)
+    df.show()
```
Comment (rahulbhatia023) [inline restonomer-core/src/test/scala/com/clairvoyant/restonomer/core/converter/CSVResponseToDataFrameConverterSpec.scala:16]: Remove show() from uni tests

File: restonomer-docs/source/restonomer_response_handler.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    body-format = "JSON"
+
+    transformations = [
+      {
+        type = "add-literal-column"
+        column-name = "col_D"
+        column-value = "val_D"
+        column-data-type = "string"
+      }]
+  }
```
Comment (rahulbhatia023) [inline restonomer-docs/source/restonomer_response_handler.md:26]: I see that in this example you are missing persistence but persistence is mandatory. This example is not accurate. I suggest to remove transformations and add persistence as transformations is optional.

File: restonomer-docs/source/restonomer_response_handler.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  data-response = {
+    body-format = "CSV"
+
+    transformations = [
+      {
+        type = "add-literal-column"
+        column-name = "col_D"
+        column-value = "val_D"
+        column-data-type = "string"
+      }]
```
Comment (rahulbhatia023) [inline restonomer-docs/source/restonomer_response_handler.md:51]: Same here: I see that in this example you are missing persistence but persistence is mandatory. This example is not accurate. I suggest to remove transformations and add persistence as transformations is optional.

**Review by rahulbhatia023 — CHANGES_REQUESTED** [review]
Suggested few changes.

File: restonomer-core/src/it/resources/mock_data/converter/csv/__files/sample_csv_response_body.csv (hunk 1 above)
Comment (umang-clairvoyant) [reply restonomer-core/src/it/resources/mock_data/converter/csv/__files/sample_csv_response_body.csv]: resolved

File: restonomer-core/src/test/scala/com/clairvoyant/restonomer/core/converter/CSVResponseToDataFrameConverterSpec.scala (hunk 2 above)
Comment (umang-clairvoyant) [reply restonomer-core/src/test/scala/com/clairvoyant/restonomer/core/converter/CSVResponseToDataFrameConverterSpec.scala]: resolved

File: restonomer-docs/source/restonomer_response_handler.md (hunk 3 above)
Comment (umang-clairvoyant) [reply restonomer-docs/source/restonomer_response_handler.md:24]: resolved

File: restonomer-docs/source/restonomer_response_handler.md (hunk 4 above)
Comment (umang-clairvoyant) [reply restonomer-docs/source/restonomer_response_handler.md]: resolved

**Pull request merged**